src/Utils/TypeInfo.php

```diff
@@ -363,52 +363,42 @@ static function ($arg) use ($node) {
                     $inputFieldType = $inputField ? $inputField->getType() : null;
                 }
                 $this->defaultValueStack[] = $inputField && $inputField->defaultValueExists() ? $inputField->defaultValue : Utils::undefined();
-                $this->inputTypeStack[]    = Type::isInputType($inputFieldType) ? $inputFieldType : null;
+                $this->inputTypeStack[]    = $inputFieldType;
                 break;
 
             case $node instanceof EnumValueNode:
                 $enumType  = Type::getNamedType($this->getInputType());
                 $enumValue = null;
                 if ($enumType instanceof EnumType) {
-                    $enumValue = $enumType->getValue($node->value);
+                    $this->enumValue = $enumType->getValue($node->value);
                 }
                 $this->enumValue = $enumValue;
                 break;
         }
     }
 
     /**
-     * @return Type
+     * @return (Type & OutputType) | null
      */
-    public function getType()
+    public function getType() : ?OutputType
     {
-        if (! empty($this->typeStack)) {
-            return $this->typeStack[count($this->typeStack) - 1];
-        }
-
-        return null;
+        return $this->typeStack[count($this->typeStack) - 1] ?? null;
     }
 
     /**
-     * @return Type
+     * @return (Type & CompositeType) | null
```

Comment:
you don't have spaces elsewhere

```suggestion
     * @return (CompositeType&Type)|null
```

Reply:
Temporarily necessary because of Scrutinizer. I'll start investigating moving off of their lame static checker in a different branch